compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingIntoPackUnPack.cpp

```diff
@@ -339,10 +331,19 @@ static FailureOr<Operation *> lowerOpWithEncoding(
                                     operands.take_front(inputs.size()),
                                     operands.drop_front(inputs.size()));
   } else {
+    auto elemTypes = llvm::map_to_vector(
+        lhsEncoding.getElementTypes().getValue(),
+        [](Attribute a) { return a.cast<TypeAttr>().getValue(); });
     SmallVector<ReassociationIndices> ri;
-    Value newLhs = getMmt4dLhs(operands[0], op, rewriter, ri);
-    Value newRhs = getMmt4dRhs(operands[1], op, rewriter, ri);
-    Value newResult = getMmt4dResult(operands[2], op, rewriter, ri);
+    Value newLhs =
+        getMmt4dOperand(operands[0], linalgOp, rewriter, ri, elemTypes,
+                        /*operandIdx=*/0, /*isResult*/ false);
```

Comment:
So the linalgOp must be a ContractionOp. I think we can just check if operandIdx is `2` or not. Then we don't need to pass `isResult` to the function. We can add some assertions to the implementation like:

```cpp
assert(linalgOp.getNumDpsInputs() == 2);
assert(linalgOp.getNumDpsInit() == 1);
```

If not, please see below comments.

nit: should have `=` in the function argument comment. Same for below.

```suggestion
                        /*operandIdx=*/0, /*isResult=*/false);                        
```

https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments